### PR TITLE
Zeros out Debug Directory for PE32 Images built with GCC5 in Release

### PR DESCRIPTION
**Pull request by Flickdm**
# Preface

Please ensure you have read the [contribution docs](https://github.com/microsoft/mu/blob/master/CONTRIBUTING.md) prior
to submitting the pull request. In particular,
[pull request guidelines](https://github.com/microsoft/mu/blob/master/CONTRIBUTING.md#pull-request-best-practices).

## Description

On linux when building using GCC5 in RELEASE the PE DebugDirectory includes the full path to the intermittent DLL.

See:

```
00000330: 3403 0000 4e42 3130 0000 0000 0000 0000  4...NB10........
00000340: 0000 0000 2f68 6f6d 652f 7573 6572 2d78  ..../home/user-x
00000350: 782f 6d75 5f6f 656d 5f73 616d 706c 652f  x/mu_oem_sample/
00000360: 4275 696c 642f 4f65 6d50 6b67 2f52 454c  Build/OemPkg/REL
00000370: 4541 5345 5f47 4343 352f 5836 342f 4f65  EASE_GCC5/X64/Oe
00000380: 6d50 6b67 2f48 656c 6c6f 5565 6669 2f48  mPkg/HelloUefi/H
00000390: 656c 6c6f 5565 6669 2f44 4542 5547 2f48  elloUefi/DEBUG/H
000003a0: 656c 6c6f 5565 6669 2e64 6c6c 0000 0000  elloUefi.dll....
000003b0: 0000 0000 0000 0000 0000 0000 0000 0000  ................
```

For each item, place an "x" in between `[` and `]` if true. Example: `[x]`.
_(you can also check items in the GitHub UI)_

- [X] Impacts functionality?
Simply zeros out the timestamps and debug directory using GenFW
- [ ] Impacts security?
- [ ] Breaking change?
Unlikely, considering no one should be depending on the DebugDirectory
- [ ] Includes tests?
- [ ] Includes documentation?

## How This Was Tested
Built on WSL2 Linux with GCC5

## Integration Instructions
N/A

### CODE REVIEW

File: BaseTools/Conf/tools_def.template
Comment (makubacki) [inline BaseTools/Conf/tools_def.template:1]: I think this should either:

1. Go directly as a patch to edk2
2. Be set at the package/module level

If (1), it probably would be best to update the version tag at the top of the file to raise attention for anyone that may have previously been dependent on the current behavior.

Comment (os-d) [reply BaseTools/Conf/tools_def.template:1]: Agreed with Michael. I think sending directly to edk2 and getting feedback makes sense. Regardless of that outcome, if a platform wants this, they can set it in their dsc.

**Comment by kuqin12** [comment]
So, are we repurposing this PR to change the GenFw behavior now? If so, can we update the PR title and description?

**Comment by makubacki** [comment]
> So, are we repurposing this PR to change the GenFw behavior now? If so, can we update the PR title and description?

If that's the case, I suggest dropping this PR and creating a new one when the new changes are ready. That clearly indicates the original approach was abandoned and focuses conversation on the new changes in the new PR.

**Pull request closed without merging**